main.tf

```diff
@@ -51,8 +51,9 @@ resource "aws_dynamodb_table" "default" {
   write_capacity   = var.autoscale_min_write_capacity
   hash_key         = var.hash_key
   range_key        = var.range_key
-  stream_enabled   = var.enable_streams
-  stream_view_type = var.enable_streams ? var.stream_view_type : ""
+  stream_enabled   = length(var.replicas) > 0 ? true : var.enable_streams
+  stream_view_type = length(var.replicas) > 0 || var.enable_streams ? var.stream_view_type : ""
```

Comment:
I believe for global tables `stream_view_type` needs to be `NEW_AND_OLD_IMAGES` - is that something we want to configure automatically or just document and expect the user to pass in?